staging/src/k8s.io/apiserver/pkg/endpoints/handlers/delete_test.go

```diff
@@ -215,6 +219,205 @@ func TestDeleteCollection(t *testing.T) {
+			normalDeleterGot, unsafeDeleterGot := &holder{}, &holder{}
+
+			var deleter rest.GracefulDeleter = &fakeDeleter{holder: normalDeleterGot}
+			switch {
+			case test.provider && test.providerHasUnsafeDeleter:
+				deleter = &corruptObjDeleterProvider{
+					fakeDeleter:   &fakeDeleter{normalDeleterGot},
+					unsafeDeleter: &fakeDeleter{unsafeDeleterGot},
+				}
+			// provider, but it return nil
+			case test.provider:
+				deleter = &nilProvider{fakeDeleter: &fakeDeleter{normalDeleterGot}}
+			}
+
+			handler := DeleteResource(deleter, true, scope, &fakeAdmitter{})
+
+			req, err := http.NewRequest("GET", "/test", ioutil.NopCloser(strings.NewReader("")))
+			if err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+			reqInfo := &request.RequestInfo{IsResourceRequest: true}
+			req = req.WithContext(request.WithRequestInfo(req.Context(), reqInfo))
+			if test.ignore {
+				q := req.URL.Query()
+				q.Add("ignoreStoreReadErrorWithClusterBreakingPotential", "true")
+				req.URL.RawQuery = q.Encode()
+			}
+
+			recorder := httptest.NewRecorder()
+			handler.ServeHTTP(recorder, req)
+			resp := recorder.Result()
+
+			if want, got := test.statusCode, resp.StatusCode; want != got {
+				t.Errorf("expected status code: %d, but got: %d", want, got)
+			}
+			body, err := io.ReadAll(resp.Body)
+			if err != nil {
+				t.Fatalf("unexpected error reading response body: %v", err)
+			}
+			if want := test.errshouldContain; !strings.Contains(string(body), want) {
+				t.Errorf("expected response body to contain %q, but got: %s", test.errshouldContain, body)
+			}
+			if want, got := test.regularDeleterInvoked, normalDeleterGot.invoked; want != got {
+				t.Errorf("expected normal deleter to be nvoked %d times, but got: %d", want, got)
+			}
+			if want, got := test.unsafeDeleterInvoked, unsafeDeleterGot.invoked; want != got {
+				t.Errorf("expected unsafw deleter to be nvoked %d times, but got: %d", want, got)
+			}
+			// when invoked, the normal deleter should never see the ignore option enabled
+			if test.regularDeleterInvoked == 1 && normalDeleterGot.ignore != nil {
+				t.Errorf("expected IgnoreStoreReadErrorWithClusterBreakingPotential to be nil for the normal deleter, but got: %v", normalDeleterGot.ignore)
+			}
+			// when invoked, the unsafe deleter should always see the option enabled
+			if want, got := test.unsafeDeleterInvoked == 1, ptr.Deref[bool](unsafeDeleterGot.ignore, false); want && want != got {
+				t.Errorf("expected IgnoreStoreReadErrorWithClusterBreakingPotential to be %t for the unsafe deleter, but got: %t", want, got)
+			}
+		})
+	}
+}
+
+type holder struct {
+	invoked int
+	ignore  *bool
+}
+
+type fakeDeleter struct {
+	holder *holder
+}
+
+func (f *fakeDeleter) Delete(ctx context.Context, name string, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
+	f.holder.invoked++
+	f.holder.ignore = options.IgnoreStoreReadErrorWithClusterBreakingPotential
```

Comment:
Would it make sense to verify in the test that `test.ignore == holder.ignore` to make sure that the correct ignore option is propagated to the deleter? Without that check, I don't see the purpose of storing the ignore option inside the holder, we could just check `test.ignore` directly